### PR TITLE
Add table oci_mysql_backup. Closes #164

### LINKED ISSUE #164: Add table oci_mysql_db_backup
**References**
Add any related links that will help us understand the resource, including vendor documentation, related GitHub issues, and Go SDK documentation.

### PR DESCRIPTION
**Pull request by rajeshbal65**
# Integration test logs
<details>
  <summary>Logs</summary>
  
```
Test names: [ 'tests/oci_mysql_db_backup' ]
No env file present for the current environment:  staging
 Falling back to .env config
No env file present for the current environment:  staging

SETUP: tests/oci_mysql_db_backup []

PRETEST: tests/oci_mysql_db_backup

TEST: tests/oci_mysql_db_backup
Running terraform
oci_core_vcn.named_test_resource: Creating...
oci_core_vcn.named_test_resource: Creation complete after 3s [id=ocid1.vcn.oc1.ap-mumbai-1.amaaaaaa6igdexaafmqr7z54e2vouzlkiftd5lhjvxyik3e67ydmhxzqowwa]
oci_core_subnet.named_test_resource: Creating...
oci_core_subnet.named_test_resource: Creation complete after 2s [id=ocid1.subnet.oc1.ap-mumbai-1.aaaaaaaahzynq4qh4svb5ghkwbhsz2sxnexq563lo4lw5ysxmpont7tzop2a]
oci_mysql_mysql_db_system.named_test_resource: Creating...
oci_mysql_mysql_db_system.named_test_resource: Still creating... [10s elapsed]
oci_mysql_mysql_db_system.named_test_resource: Still creating... [20s elapsed]
oci_mysql_mysql_db_system.named_test_resource: Still creating... [30s elapsed]
oci_mysql_mysql_db_system.named_test_resource: Still creating... [40s elapsed]
oci_mysql_mysql_db_system.named_test_resource: Still creating... [50s elapsed]
oci_mysql_mysql_db_system.named_test_resource: Still creating... [1m0s elapsed]
oci_mysql_mysql_db_system.named_test_resource: Still creating... [1m10s elapsed]
oci_mysql_mysql_db_system.named_test_resource: Still creating... [1m20s elapsed]
oci_mysql_mysql_db_system.named_test_resource: Still creating... [1m30s elapsed]
oci_mysql_mysql_db_system.named_test_resource: Still creating... [1m40s elapsed]
oci_mysql_mysql_db_system.named_test_resource: Still creating... [1m50s elapsed]
oci_mysql_mysql_db_system.named_test_resource: Still creating... [2m0s elapsed]
oci_mysql_mysql_db_system.named_test_resource: Still creating... [2m10s elapsed]
oci_mysql_mysql_db_system.named_test_resource: Still creating... [2m20s elapsed]
oci_mysql_mysql_db_system.named_test_resource: Still creating... [2m30s elapsed]
oci_mysql_mysql_db_system.named_test_resource: Still creating... [2m40s elapsed]
oci_mysql_mysql_db_system.named_test_resource: Still creating... [2m50s elapsed]
oci_mysql_mysql_db_system.named_test_resource: Still creating... [3m0s elapsed]
oci_mysql_mysql_db_system.named_test_resource: Still creating... [3m10s elapsed]
oci_mysql_mysql_db_system.named_test_resource: Still creating... [3m20s elapsed]
oci_mysql_mysql_db_system.named_test_resource: Still creating... [3m30s elapsed]
oci_mysql_mysql_db_system.named_test_resource: Still creating... [3m40s elapsed]
oci_mysql_mysql_db_system.named_test_resource: Still creating... [3m50s elapsed]
oci_mysql_mysql_db_system.named_test_resource: Still creating... [4m0s elapsed]
oci_mysql_mysql_db_system.named_test_resource: Still creating... [4m10s elapsed]
oci_mysql_mysql_db_system.named_test_resource: Still creating... [4m20s elapsed]
oci_mysql_mysql_db_system.named_test_resource: Still creating... [4m30s elapsed]
oci_mysql_mysql_db_system.named_test_resource: Still creating... [4m40s elapsed]
oci_mysql_mysql_db_system.named_test_resource: Still creating... [4m50s elapsed]
oci_mysql_mysql_db_system.named_test_resource: Still creating... [5m0s elapsed]
oci_mysql_mysql_db_system.named_test_resource: Still creating... [5m10s elapsed]
oci_mysql_mysql_db_system.named_test_resource: Still creating... [5m20s elapsed]
oci_mysql_mysql_db_system.named_test_resource: Still creating... [5m30s elapsed]
oci_mysql_mysql_db_system.named_test_resource: Still creating... [5m41s elapsed]
oci_mysql_mysql_db_system.named_test_resource: Still creating... [5m51s elapsed]
oci_mysql_mysql_db_system.named_test_resource: Still creating... [6m1s elapsed]
oci_mysql_mysql_db_system.named_test_resource: Still creating... [6m11s elapsed]
oci_mysql_mysql_db_system.named_test_resource: Still creating... [6m21s elapsed]
oci_mysql_mysql_db_system.named_test_resource: Still creating... [6m31s elapsed]
oci_mysql_mysql_db_system.named_test_resource: Still creating... [6m41s elapsed]
oci_mysql_mysql_db_system.named_test_resource: Still creating... [6m51s elapsed]
oci_mysql_mysql_db_system.named_test_resource: Still creating... [7m1s elapsed]
oci_mysql_mysql_db_system.named_test_resource: Still creating... [7m11s elapsed]
oci_mysql_mysql_db_system.named_test_resource: Still creating... [7m21s elapsed]
oci_mysql_mysql_db_system.named_test_resource: Still creating... [7m31s elapsed]
oci_mysql_mysql_db_system.named_test_resource: Still creating... [7m41s elapsed]
oci_mysql_mysql_db_system.named_test_resource: Still creating... [7m51s elapsed]
oci_mysql_mysql_db_system.named_test_resource: Still creating... [8m1s elapsed]
oci_mysql_mysql_db_system.named_test_resource: Creation complete after 8m8s [id=ocid1.mysqldbsystem.oc1.ap-mumbai-1.aaaaaaaaefo3gszgcdibxx6ilrfa3zbezho6niqmljrpf5yiarfcvgpkcjja]
oci_mysql_mysql_backup.named_test_resource: Creating...
oci_mysql_mysql_backup.named_test_resource: Still creating... [10s elapsed]
oci_mysql_mysql_backup.named_test_resource: Still creating... [20s elapsed]
oci_mysql_mysql_backup.named_test_resource: Still creating... [30s elapsed]
oci_mysql_mysql_backup.named_test_resource: Still creating... [40s elapsed]
oci_mysql_mysql_backup.named_test_resource: Still creating... [50s elapsed]
oci_mysql_mysql_backup.named_test_resource: Still creating... [1m0s elapsed]
oci_mysql_mysql_backup.named_test_resource: Still creating... [1m10s elapsed]
oci_mysql_mysql_backup.named_test_resource: Still creating... [1m20s elapsed]
oci_mysql_mysql_backup.named_test_resource: Still creating... [1m30s elapsed]
oci_mysql_mysql_backup.named_test_resource: Still creating... [1m40s elapsed]
oci_mysql_mysql_backup.named_test_resource: Creation complete after 1m46s [id=ocid1.mysqlbackup.oc1.ap-mumbai-1.aaaaaaaakmazzbkh3h4hwt23yenu3nzf2ziej3jwltpddpxnrrgjdjrtujqa]

Apply complete! Resources: 4 added, 0 changed, 0 destroyed.

Outputs:

backup_size_in_gbs = 1
backup_type = "FULL"
creation_type = "MANUAL"
data_storage_size_in_gb = 50
db_system_id = "ocid1.mysqldbsystem.oc1.ap-mumbai-1.aaaaaaaaefo3gszgcdibxx6ilrfa3zbezho6niqmljrpf5yiarfcvgpkcjja"
description = "steampipetest2470"
display_name = "steampipetest2470"
mysql_version = "8.0.25"
region = "ap-mumbai-1"
resource_id = "ocid1.mysqlbackup.oc1.ap-mumbai-1.aaaaaaaakmazzbkh3h4hwt23yenu3nzf2ziej3jwltpddpxnrrgjdjrtujqa"
resource_name = "steampipetest2470"
retention_in_days = 365
state = "ACTIVE"
tenancy_ocid = "ocid1.tenancy.oc1..aaaaaaaahnm7gleh5soecxzjetci3yjjnjqmfkr4po3hoz4p4h2q37cyljaq"

Running SQL query: test-get-query.sql

[
  {
    "backup_type": "FULL",
    "creation_type": "MANUAL",
    "db_system_id": "ocid1.mysqldbsystem.oc1.ap-mumbai-1.aaaaaaaaefo3gszgcdibxx6ilrfa3zbezho6niqmljrpf5yiarfcvgpkcjja",
    "description": "steampipetest2470",
    "display_name": "steampipetest2470",
    "id": "ocid1.mysqlbackup.oc1.ap-mumbai-1.aaaaaaaakmazzbkh3h4hwt23yenu3nzf2ziej3jwltpddpxnrrgjdjrtujqa",
    "lifecycle_state": "ACTIVE",
    "mysql_version": "8.0.25"
  }
]
✔ PASSED

Running SQL query: test-list-query.sql

[
  {
    "backup_type": "FULL",
    "creation_type": "MANUAL",
    "db_system_id": "ocid1.mysqldbsystem.oc1.ap-mumbai-1.aaaaaaaaefo3gszgcdibxx6ilrfa3zbezho6niqmljrpf5yiarfcvgpkcjja",
    "description": "steampipetest2470",
    "display_name": "steampipetest2470",
    "id": "ocid1.mysqlbackup.oc1.ap-mumbai-1.aaaaaaaakmazzbkh3h4hwt23yenu3nzf2ziej3jwltpddpxnrrgjdjrtujqa",
    "lifecycle_state": "ACTIVE",
    "mysql_version": "8.0.25"
  }
]
✔ PASSED

Running SQL query: test-not-found-query.sql

null
✔ PASSED

Running SQL query: test-turbot-query.sql

[
  {
    "region": "ap-mumbai-1",
    "tenant_id": "ocid1.tenancy.oc1..aaaaaaaahnm7gleh5soecxzjetci3yjjnjqmfkr4po3hoz4p4h2q37cyljaq",
    "title": "steampipetest2470"
  }
]
✔ PASSED

POSTTEST: tests/oci_mysql_db_backup

TEARDOWN: tests/oci_mysql_db_backup

SUMMARY:

1/1 passed.
```
</details>

# Example query results
<details>
  <summary>Results</summary>
  
```
### Basic info

```sql
select
  display_name,
  id,
  description,
  lifecycle_state as state,
  backup_type,
  mysql_version
from
  oci_mysql_db_backup;


+-----------------------------------------+------------------------------------------------------------------------------------------------+-------------------+---------+-------------+---------------+
| display_name                            | id                                                                                             | description       | state   | backup_type | mysql_version |
+-----------------------------------------+------------------------------------------------------------------------------------------------+-------------------+---------+-------------+---------------+
| Backup - Tue, Jul 6, 2021, 04:54:45 UTC | ocid1.mysqlbackup.oc1.ap-mumbai-1.aaaaaaaabl5e5sfamo4lh2uwbolkwoekywhz7nci6udk5skujhnhyujbzuaa | testing           | ACTIVE  | FULL        | 8.0.25        |
| steampipetest8561                       | ocid1.mysqlbackup.oc1.ap-mumbai-1.aaaaaaaadn2qeeinlf457nhxznlal4rezh2ru43rjqzpy7kj32jyu26ev2qq | steampipetest8561 | DELETED | FULL        | 8.0.25        |
| steampipetest9669                       | ocid1.mysqlbackup.oc1.ap-mumbai-1.aaaaaaaanvf4m6qb6khh4po33qjje3abyjl7d27ocvqrvjgjxrj7aj52simq | steampipetest9669 | DELETED | FULL        | 8.0.25        |
| test_backup                             | ocid1.mysqlbackup.oc1.ap-mumbai-1.aaaaaaaa23gunvddr5a3gbee6rzjjs225w35xnaug6m3khxm53gcf2f3v55q | ing               | ACTIVE  | INCREMENTAL | 8.0.25        |
| steampipetest8508                       | ocid1.mysqlbackup.oc1.ap-mumbai-1.aaaaaaaasa53byremltrkkuipl2tb34pzysmuknsj6rv4srxhi24odnh2vkq | steampipetest8508 | DELETED | FULL        | 8.0.25        |
| steampipetest1165                       | ocid1.mysqlbackup.oc1.ap-mumbai-1.aaaaaaaazb6uzjkzrqsjzk4xma3n7nhhuiz3bvzkpjls4k6udvzs4wavklha | steampipetest1165 | DELETED | FULL        | 8.0.25        |
| steampipetest2470                       | ocid1.mysqlbackup.oc1.ap-mumbai-1.aaaaaaaakmazzbkh3h4hwt23yenu3nzf2ziej3jwltpddpxnrrgjdjrtujqa | steampipetest2470 | DELETED | FULL        | 8.0.25        |
| steampipetest3146                       | ocid1.mysqlbackup.oc1.ap-mumbai-1.aaaaaaaackuuc4wvk3odco3vaqewvusju3q5clnwqszpuq3nbx4n7h72e4iq | steampipetest3146 | ACTIVE  | FULL        | 8.0.25        |
+-----------------------------------------+------------------------------------------------------------------------------------------------+-------------------+---------+-------------+---------------+
```

### List manual backups

```sql
select
  display_name,
  id,
  creation_type
from
  oci_mysql_db_backup
where
  creation_type = 'MANUAL';

+-----------------------------------------+------------------------------------------------------------------------------------------------+---------------+
| display_name                            | id                                                                                             | creation_type |
+-----------------------------------------+------------------------------------------------------------------------------------------------+---------------+
| Backup - Tue, Jul 6, 2021, 04:54:45 UTC | ocid1.mysqlbackup.oc1.ap-mumbai-1.aaaaaaaabl5e5sfamo4lh2uwbolkwoekywhz7nci6udk5skujhnhyujbzuaa | MANUAL        |
| steampipetest8508                       | ocid1.mysqlbackup.oc1.ap-mumbai-1.aaaaaaaasa53byremltrkkuipl2tb34pzysmuknsj6rv4srxhi24odnh2vkq | MANUAL        |
| steampipetest2470                       | ocid1.mysqlbackup.oc1.ap-mumbai-1.aaaaaaaakmazzbkh3h4hwt23yenu3nzf2ziej3jwltpddpxnrrgjdjrtujqa | MANUAL        |
| test_backup                             | ocid1.mysqlbackup.oc1.ap-mumbai-1.aaaaaaaa23gunvddr5a3gbee6rzjjs225w35xnaug6m3khxm53gcf2f3v55q | MANUAL        |
| steampipetest3146                       | ocid1.mysqlbackup.oc1.ap-mumbai-1.aaaaaaaackuuc4wvk3odco3vaqewvusju3q5clnwqszpuq3nbx4n7h72e4iq | MANUAL        |
| steampipetest8561                       | ocid1.mysqlbackup.oc1.ap-mumbai-1.aaaaaaaadn2qeeinlf457nhxznlal4rezh2ru43rjqzpy7kj32jyu26ev2qq | MANUAL        |
| steampipetest1165                       | ocid1.mysqlbackup.oc1.ap-mumbai-1.aaaaaaaazb6uzjkzrqsjzk4xma3n7nhhuiz3bvzkpjls4k6udvzs4wavklha | MANUAL        |
| steampipetest9669                       | ocid1.mysqlbackup.oc1.ap-mumbai-1.aaaaaaaanvf4m6qb6khh4po33qjje3abyjl7d27ocvqrvjgjxrj7aj52simq | MANUAL        |
+-----------------------------------------+------------------------------------------------------------------------------------------------+---------------+
```

### List backups with retention days less than 90 days

```sql
select
  display_name,
  id,
  retention_in_days
from
  oci_mysql_db_backup
where
  retention_in_days < 90;

+--------------+------------------------------------------------------------------------------------------------+-------------------+
| display_name | id                                                                                             | retention_in_days |
+--------------+------------------------------------------------------------------------------------------------+-------------------+
| test_backup  | ocid1.mysqlbackup.oc1.ap-mumbai-1.aaaaaaaa23gunvddr5a3gbee6rzjjs225w35xnaug6m3khxm53gcf2f3v55q | 89                |
+--------------+------------------------------------------------------------------------------------------------+-------------------+
```

### Count of backups per db system

```sql
select
  db_system_id,
  count(*) as backup_count
from
  oci_mysql_db_backup
group by
  db_system_id;

+--------------------------------------------------------------------------------------------------+--------------+
| db_system_id                                                                                     | backup_count |
+--------------------------------------------------------------------------------------------------+--------------+
| ocid1.mysqldbsystem.oc1.ap-mumbai-1.aaaaaaaa5agjajb6ivytlpkp7sal37dnudz6vgzngyem3hk3y7benh7h2z6a | 1            |
| ocid1.mysqldbsystem.oc1.ap-mumbai-1.aaaaaaaa5i6il6caoayjfkbfe7eeqzple7khcipxmufenxc6zdkhqcz35azq | 2            |
| ocid1.mysqldbsystem.oc1.ap-mumbai-1.aaaaaaaa7ing3e3xx6nr26blvixwkofmap4vh2fee52pnq2m2dl27b3pbzxa | 1            |
| ocid1.mysqldbsystem.oc1.ap-mumbai-1.aaaaaaaa7mqkh3fcknfi7glitu3m5fotiyb5qfc7d7dly4kqdhjprjhrl5pa | 1            |
| ocid1.mysqldbsystem.oc1.ap-mumbai-1.aaaaaaaaefo3gszgcdibxx6ilrfa3zbezho6niqmljrpf5yiarfcvgpkcjja | 1            |
| ocid1.mysqldbsystem.oc1.ap-mumbai-1.aaaaaaaanpzwumnrvbddbktfo7ojxgmnlkiu5bcwqw32ofvkyci765mg3keq | 1            |
| ocid1.mysqldbsystem.oc1.ap-mumbai-1.aaaaaaaazem2zfjzfwak4zixmepxis2dyzfu6uxii6445htf3l25imwmrpwa | 1            |
+--------------------------------------------------------------------------------------------------+--------------+
```
```
</details>

### CODE REVIEW

File: docs/tables/oci_mysql_db_backup.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+  display_name,
+  id,
+  description,
+  lifecycle_state as state,
+  backup_type,
+  mysql_version
+from
+  oci_mysql_db_backup;
+```
+
```
Comment (bigdatasourav) [inline docs/tables/oci_mysql_db_backup.md:20]:
```suggestion
```

File: docs/tables/oci_mysql_db_backup.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+select
+  display_name,
+  id,
+  creation_type
+from
+  oci_mysql_db_backup
+where
+  creation_type = 'MANUAL';
+```
+
```
Comment (bigdatasourav) [inline docs/tables/oci_mysql_db_backup.md:34]:
```suggestion
```

File: docs/tables/oci_mysql_db_backup.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+select
+  display_name,
+  id,
+  retention_in_days
+from
+  oci_mysql_db_backup
+where
+  retention_in_days < 90;
+```
+
```
Comment (bigdatasourav) [inline docs/tables/oci_mysql_db_backup.md:48]:
```suggestion
```

File: oci-test/tests/oci_mysql_db_backup/variables.tf [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+  tenancy_ocid        = var.tenancy_ocid
+  config_file_profile = var.config_file_profile
+  region              = var.region
+}
+
+resource "oci_core_vcn" "named_test_resource" {
+    compartment_id = var.tenancy_ocid
+    cidr_block = "10.0.0.0/16"
+}
+resource "oci_core_subnet" "named_test_resource" {
```
Comment (bigdatasourav) [inline oci-test/tests/oci_mysql_db_backup/variables.tf:34]:
```suggestion

resource "oci_core_subnet" "named_test_resource" {
```

File: oci-test/tests/oci_mysql_db_backup/variables.tf [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+}
+
+output "creation_type" {
+  value = oci_mysql_mysql_backup.named_test_resource.creation_type
+}
+
+output "data_storage_size_in_gb" {
+  value = oci_mysql_mysql_backup.named_test_resource.data_storage_size_in_gb
+}
+
```
Comment (bigdatasourav) [inline oci-test/tests/oci_mysql_db_backup/variables.tf:119]: remove extra spaces

File: oci/service.go [hunk 6]
```diff
@@ -51,6 +52,7 @@ type session struct {
 	KmsManagementClient            keymanagement.KmsManagementClient
 	KmsVaultClient                 keymanagement.KmsVaultClient
 	LoggingManagementClient        logging.LoggingManagementClient
+	MysqlClient                    mysql.DbBackupsClient
```
Comment (bigdatasourav) [inline oci/service.go:55]:
```suggestion
	MysqlDbBackupClient                    mysql.DbBackupsClient
```

File: oci/service.go [hunk 7]
```diff
@@ -801,6 +803,44 @@ func dnsService(ctx context.Context, d *plugin.QueryData) (*session, error) {
 	return sess, nil
 }
 
+// mysqlService returns the service client for OCI Mysql Service
+func mysqlService(ctx context.Context, d *plugin.QueryData, region string) (*session, error) {
```
Comment (bigdatasourav) [inline oci/service.go:807]:
```suggestion
func mysqlDbBackupService(ctx context.Context, d *plugin.QueryData, region string) (*session, error) {
```

File: oci/table_oci_mssql_db_backup.go [hunk 8]
```diff
@@ -0,0 +1,308 @@
...
+			},
+			{
+				Name:        "id",
+				Description: "The OCID of the backup.",
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromCamel(),
+			},
+			{
+				Name:        "description",
+				Description: " A user-supplied description of the backup.",
```
Comment (bigdatasourav) [inline oci/table_oci_mssql_db_backup.go:43]:
```suggestion
				Description: "A user-supplied description of the backup.",
```

**Review by bigdatasourav — APPROVED** [review]
LGTM

**Review by rajlearner17 — APPROVED** [review]
LGTM

**Comment by cbruno10** [comment]
@rajeshbal65 @bigdatasourav Is this the same resource type as https://registry.terraform.io/providers/hashicorp/oci/latest/docs/resources/mysql_mysql_backup, or a different type? Also, are there any other backup types within the MySQL service, or are these the only type?

**Comment by bigdatasourav** [comment]
@cbruno10 It is the same resource type https://registry.terraform.io/providers/hashicorp/oci/latest/docs/resources/mysql_mysql_backup, also there are no other backup types in MySQL. Updated the table name as oci_mysql_backup.

**Pull request merged**